packages/core/src/components/forms/numericInput.tsx

```diff
@@ -0,0 +1,407 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import { AbstractComponent, Classes, Keys } from "../../common";
+import * as Errors from "../../common/errors";
+import { Position } from "../../common/position";
+import { HTMLInputProps, IIntentProps, IProps, removeNonHTMLProps } from "../../common/props";
+import * as Utils from "../../common/utils";
+
+import { Button } from "../button/buttons";
+import { InputGroup } from "./inputGroup";
+
+export interface INumericInputProps extends IIntentProps, IProps {
+
+    /**
+     * The position of the buttons with respect to the input field
+     * @default Position.RIGHT
+     */
+    buttonPosition?: Position.LEFT | Position.RIGHT | "none";
+
+    /**
+     * Whether the input is in a non-interactive state
+     * @default false
+     */
+    disabled?: boolean;
+
+    /** The name of icon (the part after `pt-icon-`) to render on left side of input */
+    leftIconName?: string;
+
+    /** The placeholder text in the absence of any value */
+    placeholder?: string;
+
+    /**
+     * The increment between successive values when `shift` is held
+     * @default 10
+     */
+    majorStepSize?: number;
+
+    /** The maximum value of the input */
+    max?: number;
+
+    /** The minimum value of the input */
+    min?: number;
+
+    /**
+     * The increment between successive values when `alt` is held
+     * @default 0.1
+     */
+    minorStepSize?: number;
+
+    /**
+     * The increment between successive values when no modifier keys are held
+     * @default 1
+     */
+    stepSize?: number;
+
+    /** The value to display in the input field */
+    value?: number | string;
+
+    /** The callback invoked when `enter` is pressed and when the field loses focus */
+    onConfirm?(value: string): void;
+
+    /** The callback invoked when the value changes */
+    onUpdate?(value: string): void;
```

Comment:
Reasoning: this component was designed to support not only numbers, but also evaluation of mathematical expressions - hence the decision to expose `value` as a `string`. Felt cleaner to expose value as a string always, rather than a number sometimes and a string other times.